src/util.ts

```diff
@@ -91,19 +106,65 @@ export function mapBucketPathToUrlPath(
   env: Pick<Env, 'DIRECTORY_LISTING'>
 ): string[] | undefined {
   // @TODO: Use a switch statement or a Design Pattern here
+  // I strongly dislike the workers cache api
   if (bucketPath.startsWith(DIST_PATH_PREFIX)) {
-    const path = bucketPath.substring(15);
-    return [`/dist${path}`, `/download/releases${path}`];
-  } else if (
-    bucketPath.startsWith(API_PATH_PREFIX) ||
-    bucketPath.startsWith('nodejs/docs/latest/api')
-  ) {
-    const path = bucketPath.substring(22);
-    return [`/api${path}`, `/docs/latest/api${path}`];
+    const path = bucketPath.substring(DIST_PATH_PREFIX.length);
+
+    const possibleUrlPaths = new Set<string>();
+
+    // Purge directory listing of /dist/ and /download/release/
+    possibleUrlPaths.add('/dist/');
+    possibleUrlPaths.add('/download/release/');
+
+    // Purge whatever the paths we're updating
+    possibleUrlPaths.add(`/dist${path}`);
+    possibleUrlPaths.add(`/download/release${path}`);
+
+    // Purge all of the directory listings of folders starting with 'latest'
+    //  (e.g. `/dist/latest-hydrogen`)
+    // Bit of a hack, but I think this is the best we can do. The redirects
+    //  we have in `src/constants/redirectLinks.json` will be out of date since
+    //  a new version was uploaded and thus the latest has changed. We can't
+    //  really determine the new latest here unless we run something
+    //  similar to the `scripts/update-redirect-links.js` script here.
+    const latestDirectories = getAllLatestDirectories('nodejs/release');
+
+    for (const directory of latestDirectories) {
+      possibleUrlPaths.add(`/dist${directory}`);
+      possibleUrlPaths.add(`/download/release${directory}`);
+    }
+
+    return [...possibleUrlPaths];
   } else if (bucketPath.startsWith(DOCS_PATH_PREFIX)) {
-    return [`/docs${bucketPath.substring(11)}`];
+    const path = bucketPath.substring(DOCS_PATH_PREFIX.length);
+
+    const possibleUrlPaths = new Set<string>();
+
+    // Purge directory listings for /docs/ and /download/docs/
+    possibleUrlPaths.add('/docs/');
+    possibleUrlPaths.add('/download/docs/');
+
+    possibleUrlPaths.add(`/docs${path}`);
+    possibleUrlPaths.add(`/download/docs${path}`);
+
+    if (bucketPath.includes('/api/')) {
+      // Html file, purge it
+      possibleUrlPaths.add(`/api${path.substring('api'.length)}`);
+    }
+
+    // Purge all of the directory listings of folders starting with 'latest'
+    //  (e.g. `/docs/latest`)
+    // Refer to previous call for explanation
+    const latestDirectories = getAllLatestDirectories('nodejs/docs');
+
+    for (const directory of latestDirectories) {
+      possibleUrlPaths.add(`/docs${directory}`);
+      possibleUrlPaths.add(`/download/docs${directory}`);
+    }
+
+    return [...possibleUrlPaths];
   } else if (bucketPath.startsWith(DOWNLOAD_PATH_PREFIX)) {
-    return [`/download${bucketPath.substring(6)}`];
+    return [`/download${bucketPath.substring(DOWNLOAD_PATH_PREFIX.length)}`];
```

Comment:
`/download` contains both the `/docs` folder and the `/dist` folder (called `/release` inside `/download`)...

So in theory this should also go over all the paths that encompass dist and docs.

Reply:
Yeah the idea was to catch `nodejs/release/*` and `nodejs/docs/*` paths specifically since they have different behaviors with purging the `latest-*` directories. The highlighted condition is to catch everything else (e.g. `nodejs/nightly/*`).

Reply:
Yeah if the path is starts with `/nodejs/release` we return both of urls that can access it, https://github.com/nodejs/release-cloudflare-worker/pull/49/commits/b53c11dbc3bd3b8c01400414268463c2f8d6e5ea#diff-3294a832ea2276e554177e0b3007cc2d401c082912c7fbde49fa09141bf1aed1R116-R117